lib/lib-storage/src/Upload.ts

```diff
@@ -427,6 +438,31 @@ export class Upload extends EventEmitter {
     });
   }
 
+  private __calculatePartSize(targetPartSizeBytes: number, minPartSize: number): number {
+    const calculatedPartSize = Math.floor(targetPartSizeBytes / this.MAX_PARTS);
+    return Math.max(minPartSize, calculatedPartSize);
+  }
+
+  private __validateUploadPart(dataPart: RawDataPart, partSize: number): void {
+    const actualPartSize = byteLength(dataPart.data) || 0;
+
+    // Skip validation for single-part uploads (PUT operations)
+    if (dataPart.partNumber === 1 && dataPart.lastPart) {
+      return;
+    }
+
+    if (actualPartSize === 0) {
+      throw new Error(`Content length is missing on the data for part number ${dataPart.partNumber}`);
```

Comment:
This statement implies the response header is missing, since content length usually refers to the header of that name.

But it's not related to that header. This error branch is only due to programming error on our part, right? The error message should be developer-facing, e.g.

> Error: a dataPart was generated without a measurable data chunk size.